File: 206_ReverseLinkedList.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None
#再帰のやり方もあるがまだ理解できていないので後で復習

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        prev = None 
        x= head
        while x is not None:
            temp = x.next
            x.next = prev
            prev = x
            x = temp
        head = prev
        return head
        

File: Easy/No_0206_Reverse Linked List/206_ReverseLinkedList.js
//Runtime: 40 ms, faster than 99.93% of JavaScript online submissions for Reverse Linked List.
//Memory Usage: 34.9 MB, less than 86.96% of JavaScript online submissions for Reverse Linked List.

/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    var prev = null;
    var x = head;
    while (x !== null){
        var tmp = x.next;
        x.next = prev;
        prev = x;
        x = tmp
    }
    return prev;
    
};
